vaas/vaas/metrics/prometheus.py

```diff
@@ -1,39 +1,33 @@
 import logging
 
-from datetime import timedelta
-from typing import Dict, Optional, Protocol, Union
+from typing import Dict, Union
 from django.conf import settings
 from prometheus_client import CollectorRegistry, Gauge, Summary, push_to_gateway
-from prometheus_client.metrics import MetricWrapperBase
 
 from vaas.metrics.models import Metrics
 
 logger = logging.getLogger(__name__)
 
-class CollectorProtocol(Protocol):
-    def __call__(self, name: str, documentation: str, registry: CollectorRegistry) -> MetricWrapperBase:
-        ...
 
 class PrometheusClient:
     def __init__(self) -> None:
         self.host: str = f'{settings.PROMETHEUS_GATEWAY_HOST}:{settings.PROMETHEUS_GATEWAY_PORT}'
         self.job: str = settings.PROMETHEUS_GATEWAY_JOB
-        self.metrics_bucket: Dict[str, CollectorProtocol] = {}
+        self.metrics_bucket: Dict[str, Union[Summary, Gauge]] = {}
         self.registry: CollectorRegistry = CollectorRegistry()
 
-    def get_or_create(self, name: str, type: CollectorProtocol) -> CollectorProtocol:
-        name_with_suffix: str = f'{name}_microseconds'
-        metric: Optional[CollectorProtocol] = self.metrics_bucket.get(name_with_suffix)
+    def get_or_create(self, name: str, type: Union[Summary, Gauge]) -> Union[Summary, Gauge, None]:
```

Comment:
Could you explain how the result could be of type `None`?

Reply:
It cannot. But typing is yelling out that it can. So that's wy I added.

Reply:
I believe that typing is used to provide better readability, in such a case it is misleading. I recommend rethinking how to get rid of this misleading

Reply:
Right. Thanks for the clue 🎉 
The **get_or_create** function expects **Type** of **Summary** or **Gauge** so that's was the missing part.